### ISSUE TITLE
testAtomicSemaphoreThread failures (C++11)

### ISSUE THREAD

**Issue by kahrl** [minetest/minetest#3940]
With `-std=c++11`, running the unit tests always reports a failure in `testAtomicSemaphoreThread`:

```
======== Testing module TestThreading
[PASS] testStartStopWait - 378ms
[PASS] testThreadKill - 430ms
Test assertion failed: val == num_threads * 0x10000
    at test_threading.cpp:180
[FAIL] testAtomicSemaphoreThread - 17ms
======== Module TestThreading failed (1 failures / 3 tests) - 825ms
```

The test succeeds when compiling without `-std=c++11`.

System info: Intel Core i5-760 processor (x86_64), Gentoo Linux, gcc 4.9.3 p1.4, glibc 2.21-r2
Compiler flags: `CMAKE_CXX_FLAGS=-std=c++11 CMAKE_CXX_FLAGS_DEBUG=-g CMAKE_BUILD_TYPE=Debug`
Current commit: f9a90383e14a02ab0e66e5d5c36940a4ffb54502

ping @ShadowNinja

**Comment by kahrl** [comment]
Investigating this further, if I print `val` just before the failing assertion, I get output like: `val == 17645057`.
Since this is so much larger than the expected value `262144`, I am going to assume this means that the declaration `Atomic<u32> val` doesn't initialize val to 0. If I change it to `Atomic<u32> val(0)`, the test succeeds.

**Comment by nerzhul** [comment]
Fixed by 61d1751
